ui/src/datasets.tsx

```diff
@@ -147,6 +154,14 @@ export function Datasets() {
     return true;
   };
 
+  const cohorts = useMemo(
+    () =>
+      unfilteredCohorts.filter(
```

Comment:
`unfilteredCohorts` is all cohorts the user has access to, right? Would you expect this to go away once the UI is fetching cohorts from the backend?

Reply:
No, this filtering is based on the underlay. It might still go away once the UI is fetching cohorts from the backed though.

Reply:
Can you clarify what this logic is doing then? I thought `filter` was just getting the subset of cohorts that match the current underlay name.

Reply:
Yeah, that is what it's doing. I guess I mistook your question for being about access control when this is just because the UI only operates on one cohort at a time.